tensorflow/lite/micro/kernels/unidirectional_sequence_lstm_test.cc

```diff
@@ -30,446 +30,2762 @@ namespace {
+            0.01841056, -0.32764608, -0.33027974, -0.10826075, 0.20675004,
+            0.19069612, -0.03026325, -0.54532051, 0.33003211, 0.44901288,
+            0.21193194},
+    .recurrent_to_output_weights =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm *
+               kNumOutputCifgPeepholeNoProjLayerNorm]){
+            0.41613156, 0.42610586, -0.16495961, -0.5663873, 0.30579174,
+            -0.05115908, -0.33941799, 0.23364776, 0.11178309, 0.09481031,
+            -0.26424935, 0.46261835, 0.50248802, 0.26114327, -0.43736315,
+            0.33149987},
+
+    .cell_to_input_weights = nullptr,
+    .cell_to_forget_weights =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0.47485286, -0.51955009,
+                                                     -0.24458408, 0.31544167},
+    .cell_to_output_weights =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){-0.17135078, 0.82760304,
+                                                     0.85573703, -0.77109635},
+
+    .input_gate_bias = nullptr,
+    .forget_gate_bias =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){1., 1., 1., 1.},
+    .cell_gate_bias =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0., 0., 0., 0.},
+    .output_gate_bias =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0., 0., 0., 0.},
+
+    .projection_weights = nullptr,
+    .projection_bias = nullptr,
+
+    .output_state = output_state_cifg_peephole_no_proj_layer_norm,
+    .cell_state = cell_state_cifg_peephole_no_proj_layer_norm,
+
+    .input_layer_norm_coefficients =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0.1, 0.2, 0.3, 0.5},
+    .forget_layer_norm_coefficients =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0.2, 0.2, 0.4, 0.3},
+    .cell_layer_norm_coefficients =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0.7, 0.2, 0.3, 0.8},
+    .output_layer_norm_coefficients =
+        (float[kNumCellCifgPeepholeNoProjLayerNorm]){0.6, 0.2, 0.2, 0.5},
+
+    .output = output_cifg_peephole_no_proj_layer_norm,
+    .expected_output_original =
+        (float[kNumBatchCifgPeepholeNoProjLayerNorm *
+               kSequenceLengthCifgPeepholeNoProjLayerNorm *
+               kNumOutputCifgPeepholeNoProjLayerNorm]){
+            -0.102089, 0.00653987, 0.0515139, -0.0630045, -0.173317, 0.0109206,
+            0.0903292, -0.109497, -0.23827, 0.0119514, 0.119525, -0.12748},
+    .expected_output = goldekNumOutputCifgPeepholeNoProjLayerNorm,
 };
 
-TfLiteAffineQuantization intermediate_qparam_integer_peephole[5];
-
-// Expected outputs, n_batch * sequence_length * n_output
-const int8_t expected_output_integer_peephole[n_batch_integer_peephole *
-                                              sequence_length_integer_peephole *
-                                              n_output_integer_peephole] = {
-    127,  127, -16, -21, 127, 127, 23,   127, 127,
-    -128, 127, 127, 127, 127, 127, -128, 127, 127,
-};
+template <typename T>
```

Comment:
Thanks for the change!  Looks very nice. sorry about nitting on this.  Do you think it is possible to split the structure initializations before this line into a separate cc/h file. On top of my head, new files may look like: 
a unidirectional_sequence_lstm_test_config.h/cc: contains initialization of test structures. 

The benefit of this is that future reader of this test would be able to see the real functional test code faster (right now, have to scroll down 2.7k lines).   

WDYT?

Reply:
Thanks for the suggestion. Since this included file includes a lot of variable definitions (not just declarations), it is not a typical .h header file. It probably should be a .inc file as a textually included file as mentioned in https://google.github.io/styleguide/cppguide.html#Self_contained_Headers and https://google.github.io/styleguide/cppguide.html#File_Names - please let me know you opinion on this.